### DISCUSSION TITLE
操作日记记录，异步事件监听未生效

### DISCUSSION THREAD

**Discussion by zwac00001** [zuihou/zuihou-admin-cloud#284] [01.常见问题]
操作日记记录，异步事件监听未生效

**Comment by zuihou** [comment]
SysLogListener 里面的注解改成 `@EventListener`

**Reply by zwac00001** [reply]
现在使用的版本里面 SysLogListener 的注解就是 @EventListener

**Reply by zuihou** [reply]
看 SysLogAspect 发送时间没

**Reply by zwac00001** [reply]
SysLogAspect 的发送时间，在哪里看

**Reply by zwac00001** [reply]
我跟踪了代码，执行的时候未
进入SysLogListener方法里面

**Reply by zuihou** [reply]
> 看 SysLogAspect 发送时间没

看 SysLogAspect 发送事件没，  `publishEvent(sysLogDTO);`

必须加了 `@WebLog` 才会进入SysLogAspect

```
    @Pointcut("execution(public * top.tangyh.basic.base.controller.*.*(..)) || @annotation(top.tangyh.basic.annotation.log.WebLog)")
    public void sysLogAspect() {
    }
```

**Reply by zwac00001** [reply]
事件发送了 ，注解@WebLog加了的，更像是没有异步监听事件

**Reply by zwac00001** [reply]
通过断点发现，在运行时，加入了@weblog的方法的操作日志记录下来了，并执行到了SysLogEvent里面的方法，并调用了super。但在SysLogListener 里面添加了步骤日志，并且未发现有日志输出

**Comment by zuihou** [comment]
<img width="1422" alt="image" src="https://github.com/dromara/lamp-cloud/assets/8500410/c15916d1-307d-4ecc-a629-0ef2a3954060">


在需要记录日志的服务配置上图的代码，并确保`@ConditionalOnExpression` 条件为true

**Reply by zwac00001** [reply]
图片看不见

**Reply by zuihou** [reply]
<img width="1473" alt="image" src="https://github.com/dromara/lamp-cloud/assets/8500410/30a2ea84-f5ab-414c-bf7b-b062cd192e6b">

**Reply by zwac00001** [reply]
基础类的，oauth,system,bases 的操作日志都没有记录

**Reply by zwac00001** [reply]
我刚刚 ${lamp.log.type:LOGGER} 修改为 ${lamp.log.type:DB} 操作日志记录生效了

**Reply by zuihou** [reply]
> 我刚刚 ${lamp.log.type:LOGGER} 修改为 ${lamp.log.type:DB} 操作日志记录生效了

yml 中正确配置了 lamp.log.type， 就不用改代码
